devDocs/selfSignedBuild.md

```diff
@@ -0,0 +1,70 @@
+# Creating a self-signed build
+
+These instructions are based on Microsoft documentation to [create a self-signed certificate](https://docs.microsoft.com/en-us/windows/msix/package/create-certificate-package-signing).
+
+### Prerequisites
+
+From PowerShell running as administrator, install [PKI](https://github.com/PKISolutions/PSPKI#download-and-install-powershell-pki-module-from-the-powershell-gallery-using-powershell):
+
+```ps1
+Install-Module -Name PSPKI
+```
+
+### Create a self-signed certificate
+
+Using PKI, create a self signed build with a custom name (`FriendlyName`) and publisher (`Subject`).
+Other parameters are determined by [MS docs](https://docs.microsoft.com/en-us/windows/msix/package/create-certificate-package-signing#use-new-selfsignedcertificate-to-create-a-certificate).
+
+From PowerShell:
+```ps1
+New-SelfSignedCertificate -FriendlyName "LocalNVDA" -Type Custom -Subject "CN=Test NVDA Build, O=NVDA Dev, C=US" -KeyUsage DigitalSignature -CertStoreLocation "Cert:\CurrentUser\My" -TextExtension @("2.5.29.37={text}1.3.6.1.5.5.7.3.3", "2.5.29.19={text}")
+```
+
+This should output a thumbprint. Example Output:
+```ps1
+   PSParentPath: Microsoft.PowerShell.Security\Certificate::CurrentUser\My
+
+Thumbprint                                Subject
+----------                                -------
+148CB69869B802A36B3D8D801BA8D9D0F3C1484F  CN=Test NVDA Build, O=NVDA Dev, C=US
+```
+
+### Export certificate as PFX
+
+This [method uses a password](https://docs.microsoft.com/en-us/windows/msix/package/create-certificate-package-signing#password-usage) to handle access.
+
+Using PowerShell:
+```ps1
+cd <nvdaSourceDirectory>
+$password = ConvertTo-SecureString -String <Password> -Force -AsPlainText 
+Export-PfxCertificate -cert "Cert:\CurrentUser\My\<Certificate Thumbprint>" -FilePath local.pfx -Password $password
+```
+
+### Import the certificate
+Run PowerShell as Administrator, execute [Import-PfxCertificate
+](https://docs.microsoft.com/en-us/powershell/module/pki/import-pfxcertificate):
+```ps1
+cd <nvdaSourceDirectory>
+$password = ConvertTo-SecureString -String <Password> -Force -AsPlainText
+Import-PfxCertificate -Password $password -CertStoreLocation "Cert:\LocalMachine\TrustedPublisher" -FilePath local.pfx

```

Comment:
The same not found issue occurs for me with `Import-PfxCertificate`. After some googling it turns out that availability of these functions depends on the version of Windows in use rather than version of PowerShell. While for generating certificates it is reasonable to assume developers are on an uptodate OS, the same cannot be said about importing certificate - it is sometimes necessary to test something with a self signed build on older  versions of Windows which are still supported by NVDA, where this way of importing would fail. That was IMO the main advantage of the old method described on the wiki - it is OS agnostic and still works.

Reply:
As asked in https://github.com/nvaccess/nvda/pull/13987#discussion_r940787866, have you installed the prerequisites? They may require a Windows restart to become available.

According to [PKI](https://github.com/PKISolutions/PSPKI#module-requirements):

>This module can run on any of the specified operating system:
> - Windows Server 2008 R2*/2012*/2012 R2*/2016*
> - Windows 7/8/8.1/10

Note that building NVDA is only supported on Windows 10+.

Reply:
>After some googling it turns out that availability of these functions depends on the version of Windows in use rather than version of PowerShell.

Can you provide a reference for this?

Reply:
>it is sometimes necessary to test something with a self signed build on older versions of Windows which are still supported by NVDA, where this way of importing would fail.

Perhaps I can reintroduce the steps for importing certificates via the Windows UI as an alternative. This is a fail safe method across OSs.

Reply:
According to [this TechNet question](https://social.technet.microsoft.com/Forums/office/en-US/e3de0bdc-e6a0-4906-83a1-75278cbcdff3/importpfxcertificate-question?forum=winserverpowershell) (see the second answer) these PowerShell functions are available on Windows 8 and above. Adding the gui method of importing the certificate would indeed make sure that there is a consistent method to use regardless of the version of Windows on which the certificate has to be imported.

Reply:
I've added a passing reference for Windows 7. The NVDA docs do not need to cover how to use certificate manager (or similar) in detail. The intended destination store location of the certificate is provided.